src/display/annotation_layer.js

```diff
@@ -210,27 +214,6 @@ class AnnotationElement {
       page.view[3] - data.rect[3] + page.view[1],
     ]);
 
-    if (data.hasOwnCanvas) {
```

Comment:
I could be wrong, but it seems that the `data.hasOwnCanvas`-property is effectively unused now.

Hence I wonder if we could perhaps use e.g. a `this._hasOwnCanvas`-property on the Annotation-instances, in `src/core/annotation.js`, to hold the value instead?

Reply:
It's still used in `canvas.js` in order to know that the annotation must be rendered on its own canvas!

Reply:
Yes, but that value [comes from this code](https://github.com/mozilla/pdf.js/blob/89cebcb6f9b757583bc855bf0ddbde857ed6308b/src/core/annotation.js#L872-L878), and it seems to me that you could perhaps [change this code](https://github.com/mozilla/pdf.js/blob/89cebcb6f9b757583bc855bf0ddbde857ed6308b/src/core/annotation.js#L851-L852) to:
```javascript
const isUsingOwnCanvas = 
   this._hasOwnCanvas && intent & RenderingIntentFlag.DISPLAY; 
```